technology/space-ex-tech-reverts.lua

```diff
@@ -31,10 +31,423 @@ function create_tech_unit(count, time, ingredients)
   return unit
 end
 
+local automation_science_ingredients = {"automation-science-pack"}
+local logistics_science_ingredients = {"automation-science-pack","logistic-science-pack"}
+local military_science_ingredients = {"automation-science-pack","logistic-science-pack","military-science-pack"}
 local chemical_science_ingredients = {"automation-science-pack","logistic-science-pack","chemical-science-pack"}
+local military_chemical_science_ingredients = {"automation-science-pack","logistic-science-pack","military-science-pack","chemical-science-pack"}
+local production_science_ingredients = {"automation-science-pack","logistic-science-pack","chemical-science-pack","production-science-pack"}
 local utility_science_ingredients = {"automation-science-pack","logistic-science-pack","chemical-science-pack","utility-science-pack"}
+local utility_military_science_ingredients = {"automation-science-pack","logistic-science-pack","military-science-pack","chemical-science-pack","utility-science-pack"}
+local production_utility_science_ingredients = {"automation-science-pack","logistic-science-pack","chemical-science-pack","production-science-pack","utility-science-pack"}
+local production_utility_military_science_ingredients = {"automation-science-pack","logistic-science-pack","military-science-pack","chemical-science-pack","production-science-pack","utility-science-pack"}
 
 if not settings.startup["aivech-ssx-equipment-research"].value then
+  
+  --burner inserters locked behind basic automation
+  
+  --inserters locked behind electricity
+  
+  --small electric pole locked behind electricity
+  
+  --fuel refining was locked behind its own tech - petroleum gas should be behind oil processing, the other 2 behind advanced oil
+  
+  --electronics shouldn't unlock any items, but still shows as unlocking electronic circuits (though they're unlocked from the start)
+  
+  --radar is locked behind its own tech
+  
+  local automation = data.raw.technology["automation"]
+  replace_tech_cost(automation, create_tech_unit(10,10, automation_science_ingredients), nil)
+  
+  local physical_projectile_damage4 = data.raw.technology["physical-projectile-damage-4"]
```

Comment:
I wasn't going to change the costs of infinite researches, but maybe this can be added as a separate config option from general tech reverts.